src/core/lib/channel/http_client_filter.c

```diff
@@ -332,10 +332,8 @@ static grpc_error *hc_mutate_op(grpc_exec_ctx *exec_ctx,
          */
         char *t = (char *)GRPC_SLICE_START_PTR(path_with_query_slice);
         /* safe to use strlen since base64_encode will always add '\0' */
-        size_t path_length = strlen(t) + 1;
-        *(t + path_length) = '\0';
         path_with_query_slice =
-            grpc_slice_sub(path_with_query_slice, 0, path_length);
+            grpc_slice_sub(path_with_query_slice, 0, strlen(t));
```

Comment:
There's NO guarantee that t is NULL terminated here is there?
In general we shouldn't be mixing C-style strings (which can't contain NUL's) with slices (which can).

Why can't grpc_base64_encode_core return the length of string that it processed?